avail/src/main/kotlin/avail/anvil/Styles.kt

```diff
@@ -32,605 +32,2987 @@
+ * where the repetition occurs in the first succession.
+ *
+ * @property forkTarget
+ *   The zero-based nybble offset of the target
+ *   [instruction][StyleRuleInstruction] for resumption by the new
+ *   [context][StyleRuleContext]. The offset is relative to the start of the
+ *   instruction stream.
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+class ForkN constructor(
+	private val forkTarget: Int
+): StyleRuleInstruction
+{
+	init
+	{
+		assert(forkTarget >= 1) { "fork target must be ≥ 1" }
+	}
+
+	override fun toString() = "fork to @$forkTarget"
+
+	companion object : StyleRuleInstructionCoder(0xB)
+	{
+		override fun emitOn(nybbles: NybbleOutputStream, vararg operands: Int)
+		{
+			assert(operands[0] >= 1) { "fork target must be ≥ 1" }
+			nybbles.opcode(opcode)
+			nybbles.vlq(operands[0] - 1)
+		}
+
+		override fun decodeOperands(nybbles: NybbleInputStream) =
+			ForkN(nybbles.unvlq() + 1)
+	}
+}
+
+/**
+ * Match a style classifier against the special end-of-sequence classifier,
+ * represented by the empty string. On success, [succeed][ACCEPTED] the
+ * enclosing rule; on failure, [fail][REJECTED] the enclosing [rule][StyleRule].
+ *
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+object MatchEndOfSequence: StyleRuleInstructionCoder(0xC), StyleRuleInstruction
+{
+	override fun toString() = "match end of sequence"
+	override fun decodeOperands(nybbles: NybbleInputStream) = MatchEndOfSequence
+}
+
+////////////////////////////////////////////////////////////////////////////////
+//                                 Execution.                                 //
+////////////////////////////////////////////////////////////////////////////////
+
+/**
+ * A [StyleRuleContext] represents the complete machine state of a running
+ * [StyleRule].
+ *
+ * @property rule
+ *   The [rule][StyleRule] that generated this context.
+ * @property programCounter
+ *   The zero-based nybble index of the next [instruction][StyleRuleInstruction]
+ *   to execute from the [rule][StyleRule]. This is relative to the based of the
+ *   coded instruction stream.
+ * @property state
+ *   The [execution&#32;state][StyleRuleContextState].
+ * @author Todd L Smith &lt;todd@availlang.org&gt;
+ */
+data class StyleRuleContext constructor(
+	val rule: StyleRule,
+	val programCounter: Int,
+	val state: StyleRuleContextState = PAUSED)
+{
+	/**
+	 * The literal style classifier that the [rule] will attempt to match given
+	 * this [context][StyleRuleContext], or `null` if the
+	 * [program&#32;counter][programCounter] is at the end of the
+	 * [instruction][StyleRuleInstruction] [stream][StyleRule.instructions].
+	 */
+	val literal get(): String?
+	{
+		// Find the first instruction that matches a style classifier.
+		val reader = rule.instructions.inputStream(programCounter)
+		while (true)
+		{
+			when (val opcode = reader.read())
+			{
+				MatchLiteralClassifier0.opcode,
```

Comment:
Yikes. Okay, I implemented an `enum class`called `StyleRuleInstructionOpcode` which is a member of `StyleRuleInstruction`. `StyleRuleInstructionCoder` now accepts a variant of `StyleRuleInstructionOpcode`. Both complex `when` statements that switched on the numeric opcode now `StyleRuleInstructionOpcode` instead. I verified that everything still works (scary moment, because I had to change a lot of code).